File: 24 Game/24 Game.java
// Solution for 24 Game in JAVA

class Solution {
    double epsilon = 1e-6;

    private boolean canMake24(List<Double> numbers) {
        if (numbers.size() == 1) {
            return Math.abs(numbers.get(0) - 24.0) <= epsilon;
        }
        for (int i = 0; i < numbers.size(); i++) {
            for (int j = 0; j < numbers.size(); j++) {
                if (i == j) continue;
                List<Double> remaining = new ArrayList<>();
                for (int k = 0; k < numbers.size(); k++) {
                    if (k != i && k != j) remaining.add(numbers.get(k));
                }
                double num1 = numbers.get(i);
                double num2 = numbers.get(j);
                List<Double> results = new ArrayList<>();
                results.add(num1 + num2);
                results.add(num1 - num2);
                results.add(num2 - num1);
                results.add(num1 * num2);
                if (Math.abs(num2) > 0) results.add(num1 / num2);
                if (Math.abs(num1) > 0) results.add(num2 / num1);
                for (double val : results) {
                    remaining.add(val);
                    if (canMake24(remaining)) return true;
                    remaining.remove(remaining.size() - 1);
                }
            }
        }
        return false;
    }

    public boolean judgePoint24(int[] cards) {
        List<Double> nums = new ArrayList<>();
        for (int card : cards) nums.add((double) card);
        return canMake24(nums);
    }
}

File: 24 Game/24 Game.py
# Solution for 24 Game in PY

class Solution:
    def judgePoint24(self, cards: List[int]) -> bool:
        epsilon = 1e-6

        def canMake24(nums: List[float]) -> bool:
            if len(nums) == 1:
                return abs(nums[0] - 24.0) <= epsilon
            for i in range(len(nums)):
                for j in range(len(nums)):
                    if i == j:
                        continue
                    remaining = [nums[k] for k in range(len(nums)) if k != i and k != j]
                    num1, num2 = nums[i], nums[j]
                    results = [num1 + num2, num1 - num2, num2 - num1, num1 * num2]
                    if abs(num2) > 0: results.append(num1 / num2)
                    if abs(num1) > 0: results.append(num2 / num1)
                    for val in results:
                        if canMake24(remaining + [val]):
                            return True
            return False

        return canMake24([float(x) for x in cards])
